app/main.py

```diff
@@ -1,2 +1,50 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
+
+    def get_hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_index = self.get_hash(key)
+        slot = self.table[hash_index]
+
+        for index, (existing_key, existing_value) in enumerate(slot):
+            if existing_key == key:
+                slot[index] = (key, value)
+                break
+        else:
+            slot.append((key, value))
+            self.size += 1
+
+        if self.size >= self.capacity * 2:
+            self.resize()
```

Comment:
You should optimize the __setitem__ method by checking the load factor and resizing the hash table if necessary before inserting the key-value pair